util/jsonutil/jsonutil.go

```diff
@@ -211,3 +213,44 @@ func tryExtractErrorMessage(err error) string {
 func isLikelyDetailedErrorMessage(msg string) bool {
 	return !strings.HasPrefix(msg, "request.")
 }
+
+// RawMessageExtension will call json.Compact() on every json.RawMessage field when getting marshalled.
+// All other types will be marshalled as usual
+type RawMessageExtension struct {
+	jsoniter.DummyExtension
+}
+
+// CreateEncoder substitutes the default jsoniter encoder of the json.RawMessage type with ours, that
+// calls json.Compact() before writting to the stream
+func (e *RawMessageExtension) CreateEncoder(typ reflect2.Type) jsoniter.ValEncoder {
+	if typ == jsonRawMessageType {
+		return &rawMessageCodec{}
+	}
+	return nil
+}
+
+// jsonRawMessageType is declared here so we don't have to call TypeOfPtr(&json.RawMessage{}).Elem() everytime we encode
+var jsonRawMessageType = reflect2.TypeOfPtr(&json.RawMessage{}).Elem()
+
+// rawMessageCodec implements jsoniter.ValEncoder interface so we can overshadow the default json.RawMessage Encode()
+// function with our implementation
+type rawMessageCodec struct{}
+
+// Encode is intended to bahave as de default json.RawMessage Encode() with the addition of the json.Compact() call
+func (codec *rawMessageCodec) Encode(ptr unsafe.Pointer, stream *jsoniter.Stream) {
```

Comment:
IMHO The commends for `Encode` and `IsEmpty` aren't necessary.

Reply:
Removed